WalletWasabi/WabiSabi/Models/Serialization/IStateJsonConverter.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using WalletWasabi.WabiSabi.Models.MultipartyTransaction;
+
+namespace WalletWasabi.WabiSabi.Models.Serialization
+{
+	public class IStateJsonConverter : JsonConverter
```

Comment:
nit: Raname to IRoundStateJsonConverter ?

Reply:
Ummm the problem is not this json converter's name, the problem is the `IState` interface. 

I don't want to call this converter `RoundStateJsonConverter` because that would mean that the converter works with the class called `RoundState`, what it does not. This converter converts `IState` instances (`ConstructionState` and `SigningState`).

It would be good to rename that interface to something better and then rename this converter too but `RoundStateJsonConverter` would be misleading at best.

Reply:
:+1: renaming `IState`, consistent with params would be `IMultipartyTransactionState`, maybe a bit of a mouthful, but clear.

bear in mind that when we reintroduce encryption we will have to revisit this, and I think `IState` will not be serialized directly since the parameters have to be unencrypted but the events/state transitions have to be encrypted, and we probably want to serialize the encrypted data using `IBitcoinSerializable` since it's almost entirely pieces of the TX anyway.